opendevin/core/main.py

```diff
@@ -87,28 +106,36 @@ async def run_agent_controller(
 
     # runtime and tools
     runtime_cls = get_runtime_cls(config.runtime)
+    extra_kwargs = {}
+    if isinstance(runtime_cls, ServerRuntime):
+        extra_kwargs['sandbox'] = sandbox
+        # TODO: deprecate this and accept runtime as a parameter instead
+
+    logger.info(f'Initializing runtime: {runtime_cls}')
     runtime = runtime_cls(
         config=config,
         event_stream=event_stream,
-        sandbox=sandbox,
         plugins=controller.agent.sandbox_plugins,
+        **extra_kwargs,
     )
     await runtime.ainit()
-    runtime.init_runtime_tools(
-        controller.agent.runtime_tools,
-        is_async=False,
-        runtime_tools_config=runtime_tools_config,
-    )
-
-    # browser eval specific
-    # TODO: move to a better place
-    if runtime.browser and runtime.browser.eval_dir:
-        logger.info(f'Evaluation directory: {runtime.browser.eval_dir}')
-        with open(
-            os.path.join(runtime.browser.eval_dir, 'goal.txt'), 'r', encoding='utf-8'
-        ) as f:
-            task_str = f.read()
-            logger.info(f'Dynamic Eval task: {task_str}')
+    if isinstance(runtime, ServerRuntime):
+        runtime.init_runtime_tools(
+            controller.agent.runtime_tools,
+            runtime_tools_config=runtime_tools_config,
+        )
+        # browser eval specific
+        # NOTE: This will be deprecated when we move to the new runtime
+        if runtime.browser and runtime.browser.eval_dir:
+            logger.info(f'Evaluation directory: {runtime.browser.eval_dir}')
+            with open(
+                os.path.join(runtime.browser.eval_dir, 'goal.txt'),
+                'r',
+                encoding='utf-8',
+            ) as f:
+                task_str = f.read()
+                logger.info(f'Dynamic Eval task: {task_str}')
+    # TODO: Implement this for EventStream Runtime
```

Comment:
Will this cause unexpected behavior? Do we want to add a warning message?

Reply:
If we are not running web browsing evaluation, it will not cause issues - and we should fix this when we start refactor the browsing benchmark's `run_infer`.

Reply:
OK, sounds good.